internal/pkg/service/stream/storage/model/file.go

```diff
@@ -17,14 +17,15 @@ import (
 )
 
 const (
-	FileTypeCSV = "csv"
+	FileTypeCSV = FileType("csv")
 )
 
 // File represents a file prepared in the staging storage to be imported into the target storage.
 // File consists from zero or more Slices.
 type File struct {
 	FileKey
 	Retryable
+	Deleted        bool                  `json:"-"` // internal field to mark the entity for deletion, there is no soft delete
```

Comment:
File/Slice entity have new `Deleted` field, so in the `plugin.Collection.OnFile/SliceSave` we can detect the entity deletion, and (for example) delete/rollup also its statistics.

Entities in the storage repository have no soft-delete, they aren't definitions.